azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -679,19 +681,22 @@ def _execute(self, endpoint: str, database: str, query: str, payload: io.IOBase,
 
         request_headers["x-ms-client-request-id"] = client_request_id_prefix + str(uuid.uuid4())
 
+        if self._auth_provider:
+            request_headers["Authorization"] = self._auth_provider.acquire_authorization_header()
+
         if properties is not None:
             if properties.client_request_id is not None:
                 request_headers["x-ms-client-request-id"] = properties.client_request_id
             if properties.application is not None:
                 request_headers["x-ms-app"] = properties.application
             if properties.user is not None:
                 request_headers["x-ms-user"] = properties.user
+            if properties.get_option(ClientRequestProperties.no_request_timeout_option_name, False):
+                timeout = KustoClient._mgmt_default_timeout
+            else:
+                timeout = properties.get_option(ClientRequestProperties.request_timeout_option_name, timeout)
 
-        if self._auth_provider:
-            request_headers["Authorization"] = self._auth_provider.acquire_authorization_header()
-
-        if properties:
-            timeout = properties.get_option(ClientRequestProperties.request_timeout_option_name, timeout)
+        timeout = (timeout or KustoClient._mgmt_default_timeout) + KustoClient._client_server_delta

```

Comment:
Why this "or" needed? (timeout should be set by now)

Reply:
In all existing flows that call this method, timeout is set to a value. The goal of the "or" is to account for the possibility that None will be passed, without a relevant ClientRequestProperties.